OpenSim/Common/Exception.h

```diff
@@ -47,7 +47,31 @@
 //=============================================================================
 //=============================================================================
 
-namespace OpenSim { 
+namespace OpenSim {
+
+
+/** Macros to throw OpenSim exceptions. The purpose of these macros is to avoid
+having to provide the first few arguments (which are common to all OpenSim
+exceptions) to the exception constructor in the throw statements. This also
+allows us to add more details (eg class name) later easily.                   
+
+Note -- Extra braces enclosing "if" are to avoid problems when these macros are
+called within if-else statements like:
+\code{.cpp}
+          if(<some condition>)
+              OPENSIM_THROW_IF(<arguments>)
+          else
+              <some statements>                                               
+\endcode                                                                      */
+#define OPENSIM_THROW_IF(CONDITION, EXCEPTION, ...)                  \
+    {                                                                \
+    if(CONDITION)                                                    \
+        throw EXCEPTION{__FILE__, __LINE__, __func__, __VA_ARGS__};  \
+    }
+
+#define OPENSIM_THROW(EXCEPTION, ...)                                \
+    OPENSIM_THROW_IF(true, EXCEPTION, __VA_ARGS__)
+
 
```

Comment:
I have made the change.

I don't think the "stringized" version of the condition will be helpful for the:
* reader/user of the message because it may contain some low level implementation details which are not of any benefit to the reader/user in catching/fixing the error. It will end up cluttering the error message in my opinion. 
* developers because the error message will contain the location of the throw statement and the developer can see the condition that failed.